### DISCUSSION TITLE
Unable to upload example sketch

### DISCUSSION THREAD

**Discussion by Colinsp** [witnessmenow/ESP32-Cheap-Yellow-Display#152] [General]
Can anyone help me please I am trying to get an example sketch up and running on my CYD? I get the following error message in the Arduino IDE (I have the board set as ESP32 Dev Module) and have updated the user-setup with the file from this repository.

> Connecting.....
Chip is ESP32-D0WD-V3 (revision v3.1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 08:b6:1f:b7:e9:d0
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.

A fatal error occurred: Unable to verify flash chip connection (Invalid head of packet (0xE0): Possible serial noise or corruption.).
Failed uploading: uploading error: exit status 2

>

**Comment by DoctorU** [comment]
I've been getting something similar. Are you running version 2.3.2 of the Arduino IDE? I also had some updates to other libraries. 
To be fair, I've not tried to upload to the board for several weeks. This is a real issue!

**Reply by DoctorU** [reply]
My current line of investigations is going toward `brltty` conflict with the on board CH340 UART to USB chip.

https://askubuntu.com/questions/1403705/dev-ttyusb0-not-present-in-ubuntu-22-04

**Reply by DoctorU** [reply]
Final outcome from dmesg confirmed that brltty is the issue:
```
[  725.022750] usb 1-4: new full-speed USB device number 6 using xhci_hcd
[  725.271882] usb 1-4: New USB device found, idVendor=1a86, idProduct=7523, bcdDevice= 2.64
[  725.271891] usb 1-4: New USB device strings: Mfr=0, Product=2, SerialNumber=0
[  725.271895] usb 1-4: Product: USB Serial
[  725.325882] usbcore: registered new interface driver usbserial_generic
[  725.325895] usbserial: USB Serial support registered for generic
[  725.326718] usbcore: registered new interface driver ch341
[  725.326729] usbserial: USB Serial support registered for ch341-uart
[  725.326744] ch341 1-4:1.0: ch341-uart converter detected
[  725.339906] usb 1-4: ch341-uart converter now attached to ttyUSB0
[  725.373118] input: PC Speaker as /devices/platform/pcspkr/input/input15
[  725.881307] input: BRLTTY 6.4 Linux Screen Driver Keyboard as /devices/virtual/input/input16
[  726.241978] usb 1-4: usbfs: interface 0 claimed by ch341 while 'brltty' sets config #1
[  726.249353] ch341-uart ttyUSB0: ch341-uart converter now disconnected from ttyUSB0
[  726.249387] ch341 1-4:1.0: device disconnected
```
Then uninstalled BRLTTY:
```
sudo apt remove brltty
```

`ttyUSB0` now appears:

![image](https://github.com/witnessmenow/ESP32-Cheap-Yellow-Display/assets/2890936/fd5835ab-cb33-45c0-8447-f810a8a752f9)

...and I can upload again!

**Reply by dana2048** [reply]
I was getting this exact error. Changing the upload speed to anything less than 921600 solves the issue for me. In Arduino IDE, Tools Menu, Upload Speed item.

**Reply by g3wyw** [reply]
Thanks  this fixed  the problem I had on my Macbook Air with Apple Silicon. Changing the speed to 115200 solved the problem and it's working again. Also removed the CH340 drivers as not needed.

**Comment by Colinsp** [comment]
Thanks for your comments so far @DoctorU unfortunately my iMac has died so I am stuck with my old laptop at the moment so at the moment I can't do any more investigation. Hopefully a few days I will have it back.

**Reply by DoctorU** [reply]
Good luck with restoring functionality, @Colinsp .
I'm not sure if my issue is the same as yours; but at least it's one more thing for you to try.
